tools/trace.py

```diff
@@ -342,10 +342,10 @@ def _generate_field_assign(self, idx):
                 if self.probe_type == "u" and expr[0:3] == "arg":
                         arg_index = int(expr[3])
                         arg_ctype = self.usdt.get_probe_arg_ctype(
-                                self.usdt_name, arg_index - 1)
+                                self.usdt_name.encode('ascii'), arg_index - 1)
                         text = ("        %s %s = 0;\n" +
                                 "        bpf_usdt_readarg(%s, ctx, &%s);\n") \
-                                % (arg_ctype, expr, expr[3], expr)
+                                % (arg_ctype.decode(), expr, expr[3], expr)
```

Comment:
`encode()` and `decode()` mechanism is not really recommented. See details in #1582.
I think you need to change src/python/bcc/usdt.py and remove `encode()` and `decode()` there
and plus some changes in `trace.py` and you may get it working. Thanks!

Reply:
Yeah, makes sense to do it in that file better, thanks!! 

I added a commit that moves the `encode()` and `decode()` there. This way other users of `usdt.py` don't have to worry about the encoding. I've also decided to do it this way because it's how other encoding conversions are handled there. What do you think? (I would squash the commits if this gets approved :smile:)